backend/service/topology/cache.go

```diff
@@ -23,38 +23,64 @@ func (c *client) acquireTopologyCacheLock(ctx context.Context) {
 	advisoryLockId := convertLockIdToAdvisoryLockId(topologyCacheLockId)
 	ticker := time.NewTicker(time.Second * 10)
 
-	// We create our own connection to use for acquiring the advisory lock
-	// If the connection is severed for any reason the advisory lock will automatically unlock
-	conn, err := c.db.Conn(ctx)
-	if err != nil {
-		c.log.Fatal("Unable to connect to the database", zap.Error(err))
-	}
-	defer conn.Close()
```

Comment:
i think it would be simpler to open and close a connection on each loop than it is to try to maintain a conn on the client struct and figure out if it's viable, reopen it, etc

Reply:
Yeah let me give that a try

Reply:
yeah this is much simpler let me commit the refactor